add support for utf-8 formatted zip files

notspiff opened the pull request:
added in zip specification v6.3.0.
note: no handling of extra field UTF-8/MAC et al as the specifications only says this will be flagged
in the extra data field without giving details. if somebody wants this, i need samples.

i noticed this looking into http://forum.kodi.tv/showthread.php?tid=304577

https://pkware.cachefly.net/webdocs/APPNOTE/APPNOTE-6.3.0.TXT

notspiff committed:
add support for utf-8 formatted zip files

added in zip specification v6.3.0.
note: no handling of extra field UTF-8/MAC et al
as the specifications only says this will be flagged
in the extra data field without giving details.

hudokkow commented:
jenkins build this please

MartijnKaijser commented:
i can create/provide some broken .zip package if you are interested

MartijnKaijser commented:
jenkins build this please

akva2 commented:
if by broken you mean 'files made on a mac that will show you what value sits in the extra field', then yes please. put some funky chars in there to make sure it doesn't write cp467.

Paxxi commented:
I love the timing of this, got a bug report the other day about this causing addon installation failures, haven't received any sample yet though

notspiff commented:
I kinda doubt that was a coincidence see ref forum thread

23. jan. 2017 18:59 skrev "Pär Björklund" <notifications@github.com>:

> I love the timing of this, got a bug report the other day about this
> causing addon installation failures, haven't received any sample yet though
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/11527#issuecomment-274566011>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFktYHYVxS4Z1NNaWiYcUcS0Vhnel-flks5rVOpwgaJpZM4LqhFy>
> .
>

Paxxi commented:
Ahh yes, that's what I get for not following the links

notspiff committed:
if it is not utf-8, make it CP437, end of story.

this is what is supported by the spec.
makes files created by windows tools work.
NOTE: this may break other, out-of-spec files
but i suspect windows-created files are more common
than files with broken encodings.

notspiff commented:
after a bit more inspection and reading; the windows created files now work. small cost: files from < 2007 might break. i know several tools used to violate the spec. a fair cost i think.

MartijnKaijser commented:
Jenkins build this please

taxigps commented:
@notspiff Chinese filename in zip file display wrong with this pr.  it works well before.
